pkg/kubernetes/topology.js

```diff
@@ -57,8 +57,22 @@ define([
                 var all = client.select();
                 client.track(all);
                 $(all).on("changed", digest);
+                $(all).on("updated", entity_updated);
+                $(all).on("removed", entity_removed);
                 digest();
 
+                function entity_updated(ev, entity, key, last) {
+                    if($scope.selected && $scope.selected.kind === last.kind && $scope.selected.key === last.key) {
+                        $scope.selected = last;
+                    }
+                }
+
+                function entity_removed(ev, entity, key, last) {
+                    if($scope.selected && last == null) {
```

Comment:
I think this condition might look something like:

```
if ($scope.selected === entity)
    $scope.selected = null;
```

Does that work? Also there is no 'last' argument to the "removed" signal.